homeassistant/components/isy994/helpers.py

```diff
@@ -0,0 +1,250 @@
+            hass.data[ISY994_NODES][single_platform].append(node)
+            return True
+    else:
+        platforms = SUPPORTED_PLATFORMS if not single_platform else [single_platform]
+        for platform in platforms:
+            if node_uom == set(NODE_FILTERS[platform]["states"]):
+                hass.data[ISY994_NODES][platform].append(node)
+                return True
+
+    return False
+
+
+def _is_sensor_a_binary_sensor(hass: HomeAssistantType, node) -> bool:
+    """Determine if the given sensor node should be a binary_sensor."""
+    if _check_for_node_def(hass, node, single_platform=BINARY_SENSOR):
+        return True
+    if _check_for_insteon_type(hass, node, single_platform=BINARY_SENSOR):
+        return True
+
+    # For the next two checks, we're providing our own set of uoms that
+    # represent on/off devices. This is because we can only depend on these
+    # checks in the context of already knowing that this is definitely a
+    # sensor device.
+    if _check_for_uom_id(
+        hass, node, single_platform=BINARY_SENSOR, uom_list=["2", "78"]
+    ):
+        return True
+    if _check_for_states_in_uom(
+        hass, node, single_platform=BINARY_SENSOR, states_list=["on", "off"]
+    ):
+        return True
+
+    return False
+
+
+def _categorize_nodes(
+    hass: HomeAssistantType, nodes, ignore_identifier: str, sensor_identifier: str
+) -> None:
+    """Sort the nodes to their proper platforms."""
+    for (path, node) in nodes:
+        ignored = ignore_identifier in path or ignore_identifier in node.name
+        if ignored:
+            # Don't import this node as a device at all
+            continue
+
+        if isinstance(node, Group):
+            hass.data[ISY994_NODES][ISY_GROUP_PLATFORM].append(node)
+            continue
+
+        if sensor_identifier in path or sensor_identifier in node.name:
+            # User has specified to treat this as a sensor. First we need to
+            # determine if it should be a binary_sensor.
+            if _is_sensor_a_binary_sensor(hass, node):
+                continue
+
+            hass.data[ISY994_NODES][SENSOR].append(node)
+            continue
+
+        # We have a bunch of different methods for determining the device type,
+        # each of which works with different ISY firmware versions or device
+        # family. The order here is important, from most reliable to least.
+        if _check_for_node_def(hass, node):
+            continue
+        if _check_for_insteon_type(hass, node):
+            continue
+        if _check_for_uom_id(hass, node):
+            continue
+        if _check_for_states_in_uom(hass, node):
+            continue
+
+
+def _categorize_programs(hass: HomeAssistantType, programs: dict) -> None:
+    """Categorize the ISY994 programs."""
+    for platform in SUPPORTED_PROGRAM_PLATFORMS:
+        try:
+            folder = programs[KEY_MY_PROGRAMS][f"HA.{platform}"]
+        except KeyError:
+            pass
```

Comment:
```suggestion
            continue
```

Can same some indentation here by making this a `continue` and moving the below block over

Reply:
Updated.